src/Runner.Worker/StepsRunner.cs

```diff
@@ -98,124 +98,139 @@ public async Task RunAsync(IExecutionContext jobContext)
                         envContext[pair.Key] = new StringContextData(pair.Value ?? string.Empty);
                     }
 
+                    bool evaluateStepEnvFailed = false;
                     if (step is IActionRunner actionStep)
                     {
                         // Set GITHUB_ACTION
                         step.ExecutionContext.SetGitHubContext("action", actionStep.Action.Name);
 
-                        // Evaluate and merge action's env block to env context
-                        var templateEvaluator = step.ExecutionContext.ToPipelineTemplateEvaluator();
-                        var actionEnvironment = templateEvaluator.EvaluateStepEnvironment(actionStep.Action.Environment, step.ExecutionContext.ExpressionValues, step.ExecutionContext.ExpressionFunctions, VarUtil.EnvironmentVariableKeyComparer);
-                        foreach (var env in actionEnvironment)
+                        try
```

Comment:
the diff is not good, the real change is adding `try..catch` around `EvaluateStepEnvironment` and fail the step when there is an exception.

Step runner should never blow up with exception back to JobRunner.